src/NuGet.Core/NuGet.Packaging/GlobalSuppressions.cs

```diff
@@ -233,3 +233,7 @@
 [assembly: SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "The method that raised the warning returns an IDisposable object that wraps local disposable object", Scope = "member", Target = "~M:NuGet.Packaging.NuGetExtractionFileIO.DotnetCoreCreateFile(System.String)~System.IO.FileStream")]
 [assembly: SuppressMessage("Security", "IA5352:Do Not Misuse Cryptographic APIs ", Justification = "A self-issued certificate cannot be revoked.", Scope = "member", Target = "~M:NuGet.Packaging.Signing.CertificateUtility.IsSelfIssued(System.Security.Cryptography.X509Certificates.X509Certificate2)~System.Boolean")]
 [assembly: SuppressMessage("Security", "IA5352:Do Not Misuse Cryptographic APIs ", Justification = "NuGet builds a certificate chain for a certificate potentially several times. Sometimes the goal is to simply have a complete chain ignoring trust. Ultimately NuGet does build the certificate chain again with the default of RevocationMode.Online later.", Scope = "member", Target = "~M:NuGet.Packaging.Signing.SignatureUtility.GetCertificateChain(System.Security.Cryptography.X509Certificates.X509Certificate2,System.Security.Cryptography.X509Certificates.X509Certificate2Collection,System.Boolean)~NuGet.Packaging.Signing.IX509CertificateChain")]
+[assembly: SuppressMessage("Globalization", "CA2101:Specify marshaling for P/Invoke string arguments", Justification = "This call does not have strings as input or output", Scope = "member", Target = "~M:NuGet.Packaging.Signing.NativeMethods.CryptMsgOpenToEncode(NuGet.Packaging.Signing.CMSG_ENCODING,System.UInt32,System.UInt32,NuGet.Packaging.Signing.CMSG_SIGNED_ENCODE_INFO@,System.String,System.IntPtr)~NuGet.Packaging.Signing.SafeCryptMsgHandle")]
```

Comment:
Given that there's only 5 suppressions, wouldn't it be easier to address them immediately?

Reply:
My first attempt breaks a lot of tests: 840/1071 Linux tests breaks assuming Unicode on all P/Invoke calls.

Build: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6697008&view=ms.vss-test-web.build-test-results-tab
Code: https://github.com/NuGet/NuGet.Client/compare/dev-dominofire-ca2101

Reply:
That's concerning. 

https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute.charset?view=net-6.0#remarks suggests we should be using Ansi instead of unicode?

We only make a certain number of calls, so we don't really have to worry about the charset too much,

Reply:
Ok, I will test ANSI encoding.